packages/app/src/templates/components/templates-list/templates-list.component.tsx

```diff
@@ -47,45 +52,63 @@ export const TemplatesList: FunctionComponent<TemplatesListProps> = ({
   return (
     <Table role="list" hide hideableColumnsIndexes={[2, 3, 4]}>
       {templates.length > 0 ? (
-        templates.map((template) => (
-          <Row key={template.id} role="listitem">
-            <Col />
-            <Col>
-              <IconWrapper>
-                <TemplateIcon type={IconType.Template} width={3} height={3} />
-              </IconWrapper>
-            </Col>
-            <Text displayStyle={TextDisplayStyle.Paragraph1}>
-              {template.text}
-            </Text>
-            <Col>
-              <Actions>
-                <Dropdown
-                  toggler={
-                    <IconButtonWithSecondaryTooltip
-                      iconType={IconType.More}
-                      description={messages.dropdownTogllerTooltipDescription}
-                      // FIXME: The position based on offset is a sticky. However, this is a quick workaround
-                      //  for buggy overridePosition lib feature
-                      place={ElementWithTooltipPlace.Bottom}
-                      offset={{ left: 15, bottom: 5 }}
+        templates.map((template) => {
+          const { selected, indeterminate } = getRowStatus(template)
+          const handleCheckboxChange = () => toggleRow(template)
+          return (
+            <Row key={template.id} role="listitem">
+              <Col />
+              <Col>
+                <Checkbox
+                  checked={selected}
+                  onChange={handleCheckboxChange}
+                  size={Size.Large}
+                  indeterminate={indeterminate}
+                  visible={!noneRowsSelected}
+                  data-testid="template-checkbox"
```

Comment:
Shouldn't `data-testid="template-checkbox"` and `data-testid="dropdown-delete"` go to some const?